### PR TITLE
soc: nxp_imx: rt1064: Ethernet bringup

### PR DESCRIPTION
**Pull request by loicpoulain**
Enable on-board flash (4MB) (flexspi2).
Enable ethernet support.
Fix missing 4MB ARM/MPU REGION_FLASH_SIZE.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

### CODE REVIEW

File: boards/arm/mimxrt1064_evk/Kconfig.defconfig [hunk 1]
```diff
@@ -10,6 +10,14 @@ if BOARD_MIMXRT1064_EVK
 config BOARD
 	default "mimxrt1064_evk"
 
+choice CODE_LOCATION
+        default CODE_FLEXSPI2
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1064_evk/Kconfig.defconfig]: How are you programming the code to flash? Last I checked, JLink didn't support this yet.

@jhqian Did Segger resolve this issue recently?

File: boards/arm/mimxrt1064_evk/mimxrt1064_evk_defconfig [hunk 2]
```diff
@@ -14,3 +14,4 @@ CONFIG_SERIAL=y
 CONFIG_CORTEX_M_SYSTICK=y
 CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=600000000
+CONFIG_ARM_MPU=y
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1064_evk/mimxrt1064_evk_defconfig]: This change isn't needed after commit ee3f7d629b9d3f2fae06351bfa0e2dfece1cf935

File: soc/arm/nxp_imx/rt/Kconfig.soc [hunk 3]
```diff
@@ -99,14 +99,15 @@ config SOC_MIMXRT1062
...
+	select HAS_MCUX_CACHE
 	select HAS_MCUX_CCM
 	select HAS_MCUX_ELCDIF
 	select HAS_MCUX_ENET
 	select HAS_MCUX_IGPIO
 	select HAS_MCUX_LPUART
 	select HAS_MCUX_TRNG
 	select CPU_HAS_FPU_DOUBLE_PRECISION
-	select CPU_HAS_MPU
+	select CPU_HAS_ARM_MPU
```
Comment (MaureenHelm) [inline soc/arm/nxp_imx/rt/Kconfig.soc]: This change isn't needed after commit ee3f7d629b9d3f2fae06351bfa0e2dfece1cf935

File: soc/arm/nxp_imx/rt/arm_mpu_mem_cfg.h [hunk 4]
```diff
@@ -22,6 +22,8 @@
 #define REGION_FLASH_SIZE REGION_1M
 #elif CONFIG_FLASH_SIZE == 2048
 #define REGION_FLASH_SIZE REGION_2M
+#elif CONFIG_FLASH_SIZE == 4096
+#define REGION_FLASH_SIZE REGION_4M
```
Comment (MaureenHelm) [inline soc/arm/nxp_imx/rt/arm_mpu_mem_cfg.h]: Move this change to the previous commit that added support for the internal flash.

File: soc/arm/nxp_imx/rt/Kconfig.soc (hunk 3 above)
Comment (loicpoulain) [reply soc/arm/nxp_imx/rt/Kconfig.soc]: ok thanks

File: soc/arm/nxp_imx/rt/arm_mpu_mem_cfg.h (hunk 4 above)
Comment (loicpoulain) [reply soc/arm/nxp_imx/rt/arm_mpu_mem_cfg.h]: ok

File: boards/arm/mimxrt1064_evk/Kconfig.defconfig (hunk 1 above)
Comment (loicpoulain) [reply boards/arm/mimxrt1064_evk/Kconfig.defconfig]: > How are you programming the code to flash? Last I checked, JLink didn't support this yet.

I use drag-and-drop method for now.

Comment (MaureenHelm) [reply boards/arm/mimxrt1064_evk/Kconfig.defconfig]: Let's keep the defaults (both code and data) in TCM until the tools can properly support flash programming.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for updating the docs and board yaml along with the code change!

**Review by dbkinder — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Thanks @loicpoulain. I see you changed the code default to ITCM, but will you also please change the data default to DTCM?

**Comment by loicpoulain** [comment]
> Thanks @loicpoulain. I see you changed the code default to ITCM, but will you also please change the data default to DTCM?

Yes done, thx.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**